api-server/scanner-lib/src/sync/tests.rs

```diff
@@ -1207,3 +1216,531 @@ async fn check_all_destinations_are_tracked(#[case] seed: Seed) {
+
+        // check all delegations from the node are contained in the scanner
+        for (id, share) in &node_delegations {
+            let scanner_delegation = scanner_delegations.get(id).unwrap();
+            // check the shares are the same
+            assert_eq!(share, scanner_delegation.balance());
+        }
+
+        // delegations that have not been staked yet are stored in the scanner but not in the node
+        for (id, scanner_delegation) in scanner_delegations {
+            let share = node_delegations.get(&id).unwrap_or(&Amount::ZERO);
+            // check the shares are the same
+            assert_eq!(share, scanner_delegation.balance());
+        }
+    } else {
+        // the pool has been decommissioned
+        assert_eq!(Amount::ZERO, scanner_data.pledge_amount());
+    }
+}
+
+async fn check_delegations(
+    tf: &TestFramework,
+    local_state: &BlockchainState<TransactionalApiServerInMemoryStorage>,
+    delegations: &BTreeSet<DelegationId>,
+) {
+    for delegation_id in delegations {
+        check_delegation(tf, local_state, *delegation_id).await
+    }
+}
+
+async fn check_delegation(
+    tf: &TestFramework,
+    local_state: &BlockchainState<TransactionalApiServerInMemoryStorage>,
+    delegation_id: DelegationId,
+) {
+    let tx = local_state.storage().transaction_ro().await.unwrap();
+    let scanner_data = tx.get_delegation(delegation_id).await.unwrap().unwrap();
+
+    if let Some(node_data) = tf.chainstate.get_stake_delegation_data(delegation_id).unwrap() {
+        assert_eq!(node_data.source_pool(), scanner_data.pool_id());
+        assert_eq!(
+            node_data.spend_destination(),
+            scanner_data.spend_destination()
+        );
+
+        // check delegation balances are the same
+        let node_delegation_balance = tf
+            .chainstate
+            .get_stake_delegation_balance(delegation_id)
+            .unwrap()
+            .unwrap_or(Amount::ZERO);
+        assert_eq!(node_delegation_balance, *scanner_data.balance());
+    } else {
+        // the pool has been decommissioned
+        assert_eq!(Amount::ZERO, *scanner_data.balance());
+    }
+}
+
+async fn check_tokens(
+    tf: &TestFramework,
+    local_state: &BlockchainState<TransactionalApiServerInMemoryStorage>,
+    token_ids: &BTreeSet<TokenId>,
+) {
+    for token_id in token_ids {
+        check_token(tf, local_state, *token_id).await;
+    }
+}
+
+async fn check_token(
+    tf: &TestFramework,
+    local_state: &BlockchainState<TransactionalApiServerInMemoryStorage>,
+    token_id: TokenId,
+) {
+    let tx = local_state.storage().transaction_ro().await.unwrap();
+    let node_data = tf.chainstate.get_token_info_for_rpc(token_id).unwrap().unwrap();
+
+    match node_data {
+        RPCTokenInfo::FungibleToken(node_data) => {
+            let scanner_data = tx.get_fungible_token_issuance(token_id).await.unwrap().unwrap();
+
+            assert_eq!(node_data.authority, scanner_data.authority);
```

Comment:
let's add convertion function like `From<TokenInfo> for RPCTokenInfo` and compare structs. Otherwise it's easy to forget to update this test when new field is added.

same for NFT